test/performance/perf_test.lua

```diff
@@ -0,0 +1,467 @@
+    id = id + 1
+    return id
+end
+
+local function reset_gen()
+    id = 0
+end
+
+g.before_all(function(g)
+    g.cluster = helpers.Cluster:new({
+        datadir = fio.tempdir(),
+        server_command = helpers.entrypoint('srv_ddl'),
+        use_vshard = true,
+        replicasets = {
+            {
+                uuid = helpers.uuid('a'),
+                alias = 'router',
+                roles = { 'crud-router' },
+                servers = {
+                    { instance_uuid = helpers.uuid('a', 1), alias = 'router' },
+                },
+            },
+            {
+                uuid = helpers.uuid('b'),
+                alias = 's-1',
+                roles = { 'customers-storage', 'crud-storage' },
+                servers = {
+                    { instance_uuid = helpers.uuid('b', 1), alias = 's1-master' },
+                    { instance_uuid = helpers.uuid('b', 2), alias = 's1-replica' },
+                },
+            },
+            {
+                uuid = helpers.uuid('c'),
+                alias = 's-2',
+                roles = { 'customers-storage', 'crud-storage' },
+                servers = {
+                    { instance_uuid = helpers.uuid('c', 1), alias = 's2-master' },
+                    { instance_uuid = helpers.uuid('c', 2), alias = 's2-replica' },
+                },
+            },
+            {
+                uuid = helpers.uuid('d'),
+                alias = 's-2',
+                roles = { 'customers-storage', 'crud-storage' },
+                servers = {
+                    { instance_uuid = helpers.uuid('d', 1), alias = 's3-master' },
+                    { instance_uuid = helpers.uuid('d', 2), alias = 's3-replica' },
+                },
+            }
+        },
+    })
+    g.cluster:start()
+
+    g.router = g.cluster:server('router').net_box
+
+    g.router:eval([[
+        rawset(_G, 'crud', require('crud'))
+    ]])
+
+    -- Run real perf tests only with flag, otherwise run short version
+    -- to test compatibility as part of unit/integration test run.
+    g.perf_mode_on = os.getenv('PERF_MODE_ON')
+
+    g.total_report = {}
+end)
+
+g.before_each(function(g)
+    helpers.truncate_space_on_cluster(g.cluster, 'customers')
+    reset_gen()
+end)
+
+local function normalize(s, n)
+    if type(s) == 'number' then
+        s = ('%.2f'):format(s)
+    end
+
+    local len = s:len()
+    if len > n then
+        return s:sub(1, n)
+    end
+
+    return (' '):rep(n - len) .. s
+end
+
+local row_name = {
+    insert = 'insert',
+    select_pk = 'select by pk',
+    select_gt_pk = 'select gt by pk (limit 10)',
+}
+
+local column_name = {
+    without_stats_wrapper = 'without stats wrapper',
+    stats_disabled = 'stats disabled',
+    local_stats = 'local stats',
+    metrics_stats = 'metrics stats (no quantiles)',
+    metrics_quantile_stats = 'metrics stats (with quantiles)',
+}
+
+local function visualize_section(total_report, name, comment, section, params)
```

Comment:
Raw idea: visualize tables in the format that is pasteable to GitHub.

Reply:
Idea implemented